### PR TITLE
Fix[Tailwind]: Correct content purge paths

### LINKED ISSUE #1: Styles not purged in Next.js
- `windmill-react-ui` version: `0.7.4`

### What happened:

Found out that in my Next.js app *some* Windmill styles were being purged away (aka tree-shaked) by `postcss` for some unknown reason **ONLY in Production env**

Here's a visual example

Screen Shot 2022-08-18 at 12 17 03

After a lot of digging I found out that windmill-specific styles weren't being purged properly. This issue opened in another windmill fork gave me the final hint: https://github.com/ngduc/windmill-react-ui-kit/issues/1

Related to these changes for Tailwind v3: https://tailwindcss.com/docs/upgrade-guide#configure-content-sources

### What you did:

So as a workaround for now in my project I included the relative path to the windmill dist files to be purged:

Screen Shot 2022-08-23 at 23 07 34

### Relevant code or config: `tailwind.config.js`

```javascript
const defaultTheme = require('tailwindcss/defaultTheme');
const windmill = require('@cast-corp/windmill-react-ui/config')

module.exports = windmill({
  content: [
    "./pages/**/*.{js,ts,jsx,tsx}",
    "./components/**/*.{js,ts,jsx,tsx}",
    "../../node_modules/@cast-corp/windmill-react-ui/dist/**/*.{js,ts,jsx,tsx}",
  ],
  darkMode: 'class',
  theme: {},
  plugins: [],
})
```

(and tangentially I suspect [darkMode has to be specified as well](https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration) to be backwards-compatible with windmill dark mode implementation)

### Suggested solution:

As far as I can tell the fix on your fork (and proposed PR) would be to more or less follow what's suggested in the other fork issue, with adjusted paths

### PR DESCRIPTION
**Pull request by benoror**
Closes https://github.com/cast-corp/windmill-react-ui/issues/1

### CODE REVIEW

**Comment by awesomeunleashed** [comment]
Hey @benoror,

This issue is actually by design, sort of. When we switched to using a fork, I wanted to keep our existing `import {} from @windmill/react-ui` statements intact in case the original library was ever updated.

So, to do that and fix the issue you're seeing, we do this in `package.json` `dependencies`:

```
"@windmill/react-ui": "npm:@cast-corp/windmill-react-ui@^0.7.4",
```

Doing it that way, the package is still installed in the `node_modules/@windmill/react-ui` folder, making the original imports and purge paths correct.

I think there are two reasonable options here:

1. Leave it as-is (other than that purge->content fix) and make a note in the README about this, or
2. Include both sets of purge paths so that the package can be installed either way.

Thoughts?

**Comment by benoror** [comment]
@awesomeunleashed ahhh, makes total sense!

I actually like the idea of preserving that backwards-compatibility with the OG package (although I am less zero hopeful there's actually gonna be a new version coming out 😅 )

I'll like option 1., can make changes to README and leave the `content` fix (although actually the merge would not interfere with the existing `purge` key tbh)

I also like option 2. as that won't be an issue for newcomers who skip the aliasing in package.json

How about the two of them? I can work on the changes

**Comment by benoror** [comment]
@awesomeunleashed ok here's what I tried:

1. Made your suggested alias change in `package.json`:

```diff
--   "@cast-corp/windmill-react-ui": "0.7.4",
++   "@windmill/react-ui": "npm:@cast-corp/windmill-react-ui@^0.7.4",
```

2. Changed all instances of

```diff
-- import { ... } from "@cast-corp/windmill-react-ui"
++ import { ... } from "@windmill/react-ui"
```

including this `require` from `tailwind.config.js`

```diff
-- const windmill = require('@cast-corp/windmill-react-ui/config')
++ const windmill = require('@windmill/react-ui/config')
```

3. And finally removed the content purge lines:

```diff
  content: [
    "./pages/**/*.{js,ts,jsx,tsx}",
    "./components/**/*.{js,ts,jsx,tsx}",
--    "../../node_modules/@cast-corp/windmill-react-ui/lib/defaultTheme.js",
--    "../../node_modules/@cast-corp/windmill-react-ui/dist/index.js",
  ],
```

---

Unfortunately I get the same styling issue afterwards (in next.js production build), if I bring the content-purge lines back (with corrected alias path) it seems to work well again:

```diff
  content: [
    "./pages/**/*.{js,ts,jsx,tsx}",
    "./components/**/*.{js,ts,jsx,tsx}",
++    "../../node_modules/@windmill/react-ui/lib/defaultTheme.js",
++    "../../node_modules/@windmill/react-ui/dist/index.js",
  ],
```

🤔 Wondering what it could be then?

**Comment by awesomeunleashed** [comment]
> 🤔 Wondering what it could be then?

Hmm. It does sound like that *should* have worked. I'll verify the classes are actually purged correctly from those files in our project (we've overridden almost all of the default theme, so it _might_ be possible that it actually hasn't been working but everything was just being picked up from the custom theme).

> including this `require` from `tailwind.config.js`
> 
> ```diff
> -- const windmill = require('@cast-corp/windmill-react-ui/config')
> ++ const windmill = require('@windmill/react-ui/config')
> ```

Just to be sure we're looking at the same thing, your `tailwind.config.js` looks something like this, correct?

```js
const windmill = require("@windmill/react-ui/config");

module.exports = windmill({
    // content, darkMode, plugins, theme, etc.
});
```

**Comment by benoror** [comment]
> Just to be sure we're looking at the same thing, your `tailwind.config.js` looks something like this, correct?

yup, here's is it:


```javascript
const defaultTheme = require('tailwindcss/defaultTheme');
const windmill = require('@windmill/react-ui/config')

module.exports = windmill({
  content: [
    "./pages/**/*.{js,ts,jsx,tsx}",
    "./components/**/*.{js,ts,jsx,tsx}",
    "../../node_modules/@windmill/react-ui/lib/defaultTheme.js",
    "../../node_modules/@windmill/react-ui/dist/index.js",
  ],
  theme: { ...  },
  plugins: [],
})

```

**Comment by awesomeunleashed** [comment]
> ```js
>     "../../node_modules/@windmill/react-ui/lib/defaultTheme.js",
>     "../../node_modules/@windmill/react-ui/dist/index.js",
> ```

Perhaps it could be because (I assume based on these paths) your `node_modules` isn't in the same folder as the `tailwind.config.js`? I haven't tested it, but if the paths currently in `config.js` are relative to `tailwind.config.js`, then I bet they wouldn't work for you even if I merged this PR.

If that is the issue, then those lines would have to be updated to always find `node_modules` correctly in order for them to work for your repo setup.

**Comment by benoror** [comment]
> Perhaps it could be because (I assume based on these paths) your `node_modules` isn't in the same folder as the `tailwind.config.js`?

That was one of my suspicions before (https://github.com/cast-corp/windmill-react-ui/issues/1#issuecomment-1226678074), let me try again with this new context

**Comment by benoror** [comment]
@awesomeunleashed It did work! I assume postcss is not smart enough to deduce paths for workspaces during the build phase. I'll keep those two purge lines as a reminder.

Thinking it twice now I don't think many people would face this issue unless they have an exotic folder structure, so I am ok closing this if you're

**Comment by awesomeunleashed** [comment]
Great, glad you got it working. I'm guessing there's a way to make those paths search for the correct `node_modules` path instead of being hardcoded, but if you're satisfied with your workaround, I'll close this.

I did add installation notes to [the README](https://github.com/cast-corp/windmill-react-ui/blob/master/README.md) with a summary of the relevant information from our discussions.

**Comment by benoror** [comment]
@awesomeunleashed you're great, thanks so much for your time!
